shard_static/models.py

```diff
@@ -1,25 +1,29 @@
 from django.db import models
 
+from shard_static.constants import ALL_SHARD_GROUP
 from shard_static.managers import ShardStaticManager
 from shard_static.mixins import ShardStaticMixin
 
 
-class StaticSyncStatus(ShardStaticMixin, models.Model):
+class BaseStaticSyncStatus(models.Model):
     static_model_key = models.CharField(max_length=64, unique=True, verbose_name='Static Model Key')
-    last_modified = models.DateTimeField(null=True, verbose_name='Last Modified', )
-
-    diffusible = False
 
+    shard_group = ALL_SHARD_GROUP
     objects = ShardStaticManager()
 
     class Meta:
-        db_table = 'static_sync_status'
+        abstract = True
 
+    def set_revision(self, shard_static_model):
+        raise NotImplementedError
 
-class BaseShardStaticModel(ShardStaticMixin, models.Model):
-    last_modified = models.DateTimeField(null=False, verbose_name='Last Modified', )
 
+class BaseShardStaticModel(ShardStaticMixin, models.Model):
     objects = ShardStaticManager()
 
     class Meta:
         abstract = True
+
+    @classmethod
+    def get_origin_data(cls, status):
```

Comment:
This function is supposed to be in the manager, not in the model.

Reply:
If this function will be in the manager, the manager is depend on the status model.
I don't want what the manager depends on the status model.

I thought that this function gets revision value and calls manager function like `cls.objects.find_by_last_modified`.

Reply:
I don't want what the manager is depend on the status model. <- why?

If you do not want to put it in the manager, how about creating a related service?